lib/internal/sync/sync_routine.dart

```diff
@@ -34,31 +31,46 @@ class SyncRoutine {
     }
   }
 
-  Future<void> syncNotes() async {
-    if (await checkOnlineStatus() != true) {
-      return;
-    }
-    await updateLists();
-    addedNotes.forEach((note) async {
-      bool result = await noteController.add(note);
-      if (result == true) {
-        saveSynced(note);
-      }
-    });
-    updatedNotes.forEach((note, delta) async {
-      bool result = await noteController.update(note.id, delta);
-      if (result == true) {
-        saveSynced(note);
-      }
-    });
-    deletedNotes.forEach((note) async {
-      bool result = await noteController.delete(note.id);
-      if (result == true) {
-        deleteSynced(note);
+  void syncNotes() {
+    checkOnlineStatus().then((status) {
```

Comment:
I usually don't trust .then(), might be better to use async/await as it doesn't really change the flow in here and can be done easily